django/db/models/query_utils.py

```diff
@@ -169,6 +174,79 @@ def _check_parent_chain(self, instance, name):
         return None
 
 
+class RegisterLookupMixin(object):
+    def _find_lookup(self, lookup_name, class_attr):
+        """
+        Find lookup or transform registered on this, or any parent, class.
+        `class_attr` can be either `class_lookups` or `class_transforms`.
+        """
+        try:
+            return getattr(self, class_attr)[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent classes.
+            for parent in inspect.getmro(self.__class__):
+                parent_lookups = getattr(parent, class_attr, None)
+                if not parent_lookups:
+                    continue
+                if lookup_name in parent_lookups:
+                    return parent_lookups[lookup_name]
+        except AttributeError:
+            # Nothing has been registered
+            pass
+        return None
+
+    def get_lookup(self, lookup_name):
+        found = self._find_lookup(lookup_name, 'class_lookups')
+        if found is None and hasattr(self, 'output_field'):
+            return self.output_field.get_lookup(lookup_name)
+        return found
+
+    def get_transform(self, lookup_name):
+        found = self._find_lookup(lookup_name, 'class_transforms')
+        if found is None and hasattr(self, 'output_field'):
+            return self.output_field.get_transform(lookup_name)
+        return found
+
+    @classmethod
+    def register_lookup(cls, lookup, lookup_name=None):
+        from django.db.models.lookups import Transform
+        if issubclass(lookup, Transform):
+            warnings.warn(
+                'Registering transforms with register_lookup is deprecated. '
+                'Use register_transform instead.',
+                deprecation.RemovedInDjango20Warning, stacklevel=2
+            )
+            return cls.register_transform(lookup, lookup_name)
+        if lookup_name is None:
+            lookup_name = lookup.lookup_name
+        if 'class_lookups' not in cls.__dict__:
+            cls.class_lookups = {}
+        cls.class_lookups[lookup_name] = lookup
+        return lookup
+
+    @classmethod
+    def register_transform(cls, transform, lookup_name=None):
+        if lookup_name is None:
+            lookup_name = transform.lookup_name
+        if 'class_transforms' not in cls.__dict__:
+            cls.class_transforms = {}
+        cls.class_transforms[lookup_name] = transform
+        return transform
+
+    @classmethod
+    def _unregister_lookup(cls, lookup, lookup_name=None):
```

Comment:
No, I don't think so. It's not public API, and should only be used for testing. If we want to open up this method as public then we should create both unregister_lookup and unregister_transform.